src/app/cli/src/init/client.ml

```diff
@@ -2169,6 +2169,216 @@ let thread_graph =
+            let chunk_amount = chunk_amounts.(i) in
+            let num_updates = List.length kps in
+            let amount_per_update = chunk_amount / num_updates in
+            let update_rem = chunk_amount mod num_updates in
+            let update_amounts =
+              Array.init num_updates ~f:(fun i ->
+                  if i = 0 then
+                    if 0 < amount_rem then amount_per_update + 1 - fee
+                    else amount_per_update - fee
+                  else if i < amount_rem then amount_per_update + 1
+                  else amount_per_update )
+            in
+            assert (
+              Array.fold update_amounts ~init:0 ~f:( + ) = chunk_amount - fee ) ;
+            if Int.is_negative update_amounts.(0) then (
+              Format.eprintf
+                "Calculated negative amount for account update; increase \
+                 amount or lower fee@." ;
+              Core.exit 1 ) ;
+            let memo_str = sprintf "ITN account funder, chunk %d" i in
+            let memo = Signed_command_memo.create_from_string_exn memo_str in
+            let multispec :
+                Transaction_snark.For_tests.Multiple_transfers_spec.t =
+              let fee = Currency.Fee.of_nanomina_int_exn fee in
+              let sender = (fee_payer_keypair, !fee_payer_current_nonce) in
+              let fee_payer = None in
+              let receivers =
+                List.mapi kps ~f:(fun j kp ->
+                    let pk =
+                      kp.public_key |> Signature_lib.Public_key.compress
+                    in
+                    let amount =
+                      update_amounts.(j) |> Currency.Amount.of_nanomina_int_exn
+                    in
+                    (pk, amount) )
+              in
+              let amount =
+                chunk_amount |> Currency.Amount.of_nanomina_int_exn
+              in
+              let zkapp_account_keypairs = [] in
+              let new_zkapp_account = false in
+              let snapp_update = Account_update.Update.dummy in
+              let actions = [] in
+              let events = [] in
+              let call_data = Snark_params.Tick.Field.zero in
+              let preconditions = None in
+              { fee
+              ; sender
+              ; fee_payer
+              ; receivers
+              ; amount
+              ; zkapp_account_keypairs
+              ; memo
+              ; new_zkapp_account
+              ; snapp_update
+              ; actions
+              ; events
+              ; call_data
+              ; preconditions
+              }
+            in
+            fee_payer_current_nonce :=
+              Account.Nonce.succ !fee_payer_current_nonce ;
+            let zkapp_command =
+              Transaction_snark.For_tests.multiple_transfers multispec
+            in
+            let%bind res =
+              Daemon_rpcs.Client.dispatch Daemon_rpcs.Send_zkapp_commands.rpc
```

Comment:
When discussing this issue, we considered sending these zkapp transactions in batches of ten.
So that one transaction pool diff would contain 10 transactions making total of 80 transfers.

Is it possible to implement this logic?

Reply:
The current code to send the txns is sending them one at a time. The procedure that puts them in the pool accepts a list, so we could send lists of 10 at a time.

Reply:
Good, let's go forward with this one (list of ten)!